test/e2e/framework.go

```diff
@@ -702,22 +722,85 @@ func (data *TestData) podWaitForRunning(timeout time.Duration, name, namespace s
 	return err
 }
 
-// podWaitForIP polls the K8s apiserver until the specified Pod is in the "running" state (or until
-// the provided timeout expires). The function then returns the IP address assigned to the Pod.
-func (data *TestData) podWaitForIP(timeout time.Duration, name, namespace string) (string, error) {
+// podWaitForIPs polls the K8s apiserver until the specified Pod is in the "running" state (or until
+// the provided timeout expires). The function then returns the IP addresses assigned to the Pod.
+func (data *TestData) podWaitForIPs(timeout time.Duration, name, namespace string) (*PodIPs, error) {
+	return data.podWaitForIPsAndCheckAF(timeout, name, namespace, true)
+}
+
+// hostNetworkPodWaitForIPs polls the K8s apiserver until the specified Pod is in the "running" state (or until
+// the provided timeout expires). The function is called if the Pod is configured with "hostNetwork". The function
+// doesn't require one IP address is assigned in each cluster supported Pod CIDR, for such Pod has only one IP,
```

Comment:
s/is assigned in each cluster supported Pod CIDR/to be assigned for  each cluster-supported Pod CIDR

Reply:
Merge this function with `podWaitForIPs`.